dcos/http.py

```diff
@@ -218,12 +214,20 @@ def request(method,
     if verify is not None:
         silence_requests_warnings()
 
-    response = _request(method, url, is_success, timeout,
-                        verify=verify, **kwargs)
+    response = _request('head', url, is_success, timeout,
+                        verify=verify)
+    i = 0
+    while i < 3 and response.status_code == 401:
+        auth_response = _request_with_auth(response, 'head', url, is_success,
```

Comment:
You don't want to overwrite the orig response given that the www-authenticate header will be lost.